src/VuMeter.cpp

```diff
@@ -0,0 +1,155 @@
+  EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+  OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+  NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+  HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+  WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+  OTHER DEALINGS IN THE SOFTWARE.
+*/
+//*****************************************************************
+
+/**
+ * \file Limiter.cpp
+ * \author Dominick Hing
+ * \date August 2022
+ * \license MIT
+ */
+
+#include "VuMeter.h"
+
+#include <QVector>
+
+#include "jacktrip_types.h"
+
+//*******************************************************************************
+void VuMeter::init(int samplingRate)
+{
+    ProcessPlugin::init(samplingRate);
+    if (samplingRate != fSamplingFreq) {
+        std::cerr << "Sampling rate not set by superclass!\n";
+        std::exit(1);
+    }
+
+    fs = float(fSamplingFreq);
+    for (int i = 0; i < mNumChannels; i++) {
+        vumeterP[i]->init(fs);
+        // int ndx = vumeterUIP[i]->getParamIndex("NumClientsAssumed");
+        // vumeterUIP[i]->setParamValue(ndx, mNumClients);
+    }
+
+    /* Set meter values to the default floor */
+    mValues.resize(mNumChannels);
+    QVector<float>::iterator it;
+    for (it = mValues.begin(); it != mValues.end(); ++it) {
+        *it = -80.0;
+    }
+
+    /* Start timer */
+    int timeout_ms = 100;
+    connect(&mTimer, &QTimer::timeout, this, &VuMeter::onTick);
+    mTimer.setTimerType(Qt::PreciseTimer);
+    mTimer.setInterval(timeout_ms);
+    mTimer.setSingleShot(false);
+    mTimer.start();
+
+    inited = true;
+}
+
+//*******************************************************************************
+void VuMeter::compute(int nframes, float** inputs, float** /*_*/)
+{
+    // Note that the second parameter is unused. This is because all of the ProcessPlugins
+    // require the same function signature for the compute() function and is normally used
+    // for the faust plugin output. However, this plugin is not supposed to modify the
+    // signal itself like the other plugins (e.g. Limiter) do, so we don't want to write
+    // to this buffer. We just need to report the VU meter output
+
+    if (not inited) {
+        std::cerr << "*** VuMeter " << this << ": init never called! Doing it now.\n";
+        if (fSamplingFreq <= 0) {
+            fSamplingFreq = 48000;
+            std::cout << "Limiter " << this
+                      << ": *** HAD TO GUESS the sampling rate (chose 48000 Hz) ***\n";
+        }
+        init(fSamplingFreq);
+    }
+
+    for (int i = 0; i < mNumChannels; i++) {
+        /* Run the signal through Faust to  */
+        QVector<float> meterBuf(nframes);
+        float* meterBufPtr = meterBuf.data();
+        float** output     = &meterBufPtr;
+        vumeterP[i]->compute(nframes, &inputs[i], output);
```

Comment:
I'm not sure why the Faust processing is necessary. If we're just taking max (peak), why not just use the `float` samples directly?

Reply:
I think it leaves us with some additional flexibility, and I was using it primarily because the original metering was RMS. If we ever wanted to customize how the metering works, we would need to use Faust. It's also nice to have some separation between the audio processing modules and the rest of the codebase. The Faust stuff can be experimented with and tested on separately.

There's also not really a performance cost. Perf test results: with bufsize 1024 and 2 channels, normal C++ would take about 30 microseconds to process the buffers, and Faust C++ takes about 45 microseconds. By comparison, including the freeverb plugin would add about 400 microseconds of compute time.

Reply:
I agree that it doesn't provide immediate benefits - it just would leave us with more flexible options later down the road if we wanted to exercise those options. But I can take it out if you feel strongly about it.